core/src/main/java/jenkins/fingerprints/FileFingerprintStorage.java

```diff
@@ -210,6 +217,61 @@ public boolean isReady() {
         return new File(Jenkins.get().getRootDir(),"fingerprints").exists();
     }
 
+    /**
+     * Perform Fingerprint cleanup.
+     */
+    public void iterateAndCleanupFingerprints(TaskListener taskListener) {
+        Object fingerprintStorage = FingerprintStorage.get();
+        if (!(fingerprintStorage instanceof FileFingerprintStorage)) {
+            logger.fine("External fingerprint storage is configured. Skipping execution");
+            return;
+        }
+
+        int numFiles = 0;
+
+        File root = new File(getRootDir(), FINGERPRINTS_DIR_NAME);
+        File[] files1 = root.listFiles(f -> f.isDirectory() && f.getName().length()==2);
```

Comment:
Why == 2?

Reply:
this code is just moved

Reply:
Yes, this is ported code. The reason is because of the structure in which fingerprints are stored on file system storage. Say the fingerprint md5 is `abc123456789` then it gets saved inside `ab`/`c1`/`23456789.xml`